salt/modules/mount.py

```diff
@@ -811,14 +818,14 @@ def rm_automaster(name, device, config='/etc/auto_salt'):
                 lines.append(line)
     except (IOError, OSError) as exc:
         msg = "Couldn't read from {0}: {1}"
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
 
     try:
         with salt.utils.files.fopen(config, 'w+') as ofile:
-            ofile.writelines(lines)
+            ofile.writelines(salt.utils.data.decode(lines))
```

Comment:
Same as above, this needs to be an `encode`, and we need to open the file with a mode of `'wb'`.